### ISSUE TITLE
SeldonDeployment creation strips out an empty "children" list field in the manifest

### ISSUE THREAD

**Issue by gsunner** [SeldonIO/seldon-core#242]
How to reproduce:
1) Create a SeldonDeployment with children as empty list in the graph.
2) Compare the original manifest  with one obtained using "kubectl get seldondeployments ..."

A diff of the two manifests will show the children field as missing.

Tools that provision deployments (eg argocd) will see the difference as an out of sync deployment.

![seldon-core-crd-issue](https://user-images.githubusercontent.com/8726172/46284156-b7787780-c56e-11e8-831a-3fd7013a555a.png)

**Comment by cliveseldon** [comment]
The core issue is the parsing and output into JSON from proto buffers stripping default fields. 

The solution would be to update the CRD to include the Status field and use the /status update endpoint available from Kubernetes 1.11 API to ensure just the Status fields are updated. We will need to wait until the k8s Java client includes this support or we add it.

For illustration the API call from the python client is : https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CustomObjectsApi.md#replace_cluster_custom_object_status

We would need to check the version of k8s running to be >= 1.11 for this to be used otherwise we would need to fallback to updating the whole CR object when updating the status. 

to check the version in java
```
VersionApi apiInstance = new VersionApi(client);
		try {
		    VersionInfo result = apiInstance.getCode();
		    System.out.println(result);
		} catch (ApiException e) {
		    System.err.println("Exception when calling VersionApi#getCode");
		    e.printStackTrace();
		}
```

**Comment by cliveseldon** [comment]
This has been merged in the k8s java client so we need to wait until a release to merge this code:
https://github.com/kubernetes-client/java/pull/381
